pkg/plugins/aws/ami/condition.go

```diff
@@ -0,0 +1,74 @@
+package ami
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+	"github.com/updatecli/updatecli/pkg/core/scm"
+)
+
+// Condition test if a image matching specific filters exist.
+func (a *AMI) Condition(source string) (bool, error) {
+
+	if source != "" {
+		// Based on source information,
+		// we try to define a default image-id resource
+		// if not researched
+		isImageIDDefined := false
+		for i := 0; i < len(a.Filters); i++ {
+			if strings.Compare(a.Filters[i].Name, "image-id") == 0 {
+				isImageIDDefined = true
+				break
+			}
+
+		}
+
+		// Set image-id to source output if not yet defined
+		if !isImageIDDefined {
+			a.Filters = append(a.Filters, Filter{
+				Name:   "image-id",
+				Values: source,
+			})
+		}
+
+	} else {
+		if len(a.Filters) == 0 {
+			logrus.Infof("\u2717 No AMI could be found as no AMI filters defined\n")
+			return false, nil
+		}
+	}
+
+	svc, errs := a.Init()
+
+	for _, err := range errs {
+		logrus.Error(err)
+	}
+	if len(errs) > 0 {
+		return false, errors.New("Too many errors")
+	}
+
+	result, err := a.GetLatestAmiID(svc)
+
+	if err != nil {
+		return false, err
+	}
+
+	if len(result) > 0 {
+		logrus.Infof("\u2714 AMI %q found\n", result)
+		return true, nil
+	}
+
+	fmt.Printf("\u2717 No AMI found matching criteria in region %s\n", a.Region)
+
+	return false, nil
+}
+
+// ConditionFromSCM is a placeholder to validate the condition interface
+func (a *AMI) ConditionFromSCM(source string, scm scm.Scm) (bool, error) {
+
+	fmt.Printf("\u2717 Condition with SCM is not supported, please remove the scm block \n")
+
+	return false, errors.New("condition with SCM is not supported")
```

Comment:
```suggestion
	fmt.Printf("\u2717 Condition with SCM is not supported, please remove the scm block \n")

	return false, errors.New("condition with SCM is not supported")
```

You might want to define a custom error and raise it (instead of printing twice and manipulating strings)

Reply:
We are  connected, I was thinking exactly the same quite recently

Reply:
It think this should be an improvement done outside this pull request as it would cover every other resources

Reply:
Not really sure (as it would be done here at least), but it's not blocking so I'm resolving the convo.